### DISCUSSION TITLE
MeshLab 2023.12 is out!

### DISCUSSION THREAD

**Discussion by alemuntoni** [cnr-isti-vclab/meshlab#1449] [General]
This version brings several bugfixes, and new plugins:

- filter_cubization (thanks @michisco)
- filter_developability (thanks @lucaceschi)
- filter_embree (thanks @PFasano99 )
- filter_mesh_alpha_wrap (thanks @ptc-lexvandersluijs)

Starting from today, MeshLab is available from the [Microsoft Store](https://apps.microsoft.com/detail/9N4TL5DXC6C0?hl=en-US&gl=US)!

**Comment by nerdCopter** [comment]
difference in `12` vs `12d`?

**Reply by alemuntoni** [reply]
Same as the older versions: 12d is built with double precision floating point (the default one used by pymeshlab).

**Reply by rotu** [reply]
I had this same question and, and it would be nice to have this documented somewhere more discoverable. Maybe on the Releases page itself (when you're trying to figure out which release asset to download) or in the [Readme](https://github.com/cnr-isti-vclab/meshlab/releases)?

**Reply by cod3monk3y** [reply]
And `12` is built with single-precision float? 64-bit integer discretization? Throwback high performance 8-bit integers? As a new user, I'm not familiar with the old version naming scheme. And it's still not entirely clear what the difference is or why someone might want a double precision vs. whatever the other one is.

**Reply by eliliam** [reply]
@cod3monk3y did you ever determine this? If not, @alemuntoni could you clarify?

**Reply by rpavlik** [reply]
single precision float. The reason you would want the "d" is for pymeshlab.

**Comment by aslze** [comment]
What is this Embree that seems to take a lot of space in the installer? I see no mention of it in the UI buttons or filter list.

**Reply by alemuntoni** [reply]
https://www.embree.org/

It is a library for ray tracing, used by the new plugin `filter_embree`.

**Reply by aslze** [reply]
I assumed that. But I don't see that filter anywhere in the UI menus, buttons, etc.

**Reply by alemuntoni** [reply]
If you search `embree` in the search bar, it should find the filters that use the library (the library is mentioned in the description of the filters). Right now there are filters like ambient occlusion and obscurance that will likely replace the older ones that run on GPU (and create lot of portability issues).

![image](https://github.com/cnr-isti-vclab/meshlab/assets/6656168/27289072-ab2b-4b96-ae40-af1bb081a515)

**Reply by aslze** [reply]
OK, I see. Thanks!

**Comment by piwawa** [comment]
How to install the release and open it on manjaro? I can't open it after extract [MeshLab2023.12-linux.tar.gz](https://github.com/cnr-isti-vclab/meshlab/releases/download/MeshLab-2023.12/MeshLab2023.12-linux.tar.gz)

It shows this:
"Could not find the program 'meshlab'"

**Reply by alemuntoni** [reply]
Did you try the appimage?

**Comment by whbwhb521** [comment]
Can provide pymeshlab-2023.12 wheels for py37 version？

**Reply by alemuntoni** [reply]
Nope. Python 3.7 reached the [end of life](https://devguide.python.org/versions/) on June 2023.
If you need pymeshlab for python 3.7, you'll need to build it by yourself.

**Comment by sngkhiem** [comment]
i build this on windows by cmake but it say -- Configuring incomplete, errors occurred!, can you help me with this
[log.txt](https://github.com/cnr-isti-vclab/meshlab/files/14821896/log.txt)

**Reply by meiyy** [reply]
The log file reports an error:

> CMake Error at src/CMakeLists.txt:88 (add_subdirectory):
>   The source directory
> 
>     D:/meshlab-main/src/vcglib
> 
>   does not contain a CMakeLists.txt file.

It seams like the source of vcglib is missing. You probably need to clone https://github.com/cnr-isti-vclab/vcglib into D:/meshlab-main/src/vcglib and then try to rebuild meshlab.

**Reply by sngkhiem** [reply]
Thanks for reply me, I fixed that isssue, but I face with new problem, can u help me

Link: https://github.com/cnr-isti-vclab/meshlab/issues/1483

**Comment by thangla123** [comment]
![snapshot03](https://github.com/user-attachments/assets/ba65b323-8dba-4ed9-b071-0b9850287626)
How can I snapshot with PymeshLab like Meshlab because when I snapshot with Pymeshlab , the Face color is always white
![output000002_hehe_1](https://github.com/user-attachments/assets/97e0a9a7-0bf8-4cb4-8b57-af8f252c584b)

**Comment by StEvUgnIn** [comment]
Why is there no update since 2023?
